src/main/index.ts

```diff
@@ -65,14 +65,23 @@ log.transports.file.format = "[{y}-{m}-{d} {h}:{i}:{s}.{ms}][{processType}][{lev
 log.eventLogger.format = "Electron event {eventSource}#{eventName} observed";
 
 log.hooks.push((message, transport) => {
-  if (transport !== log.transports.file) return message;
-
-  // Annoyingly spammed log from YTM that doesn't need to get added to the log file
-  if (message.data[0].includes("Third-party cookie will be blocked.")) return false;
-  for (let i = 0; i < message.data.length; i++) {
-    let dataString: string = message.data[i] as string;
-    dataString = dataString.replaceAll(/(?<=((https|http):\/\/)?.{1,64}(\..{1,64})?\..{1,64}\/)([\S]+)/gm, "[REDACTED]");
-    message.data[i] = dataString;
+  if (transport !== log.transports.file) { return message; }
+
+  const isSpamMessage = (data: any) => {
+    return typeof data === 'string' && data.includes("Third-party cookie will be blocked.");
+  };
+
+  if (message?.data?.[0] && isSpamMessage(message.data[0])) {
+    return false;
+  }
+
+  if (Array.isArray(message.data)) {
```

Comment:
As mentioned in my comment, this feels redundent given that this is always an array, and type checking for the message.data shouldn't be needed?

The contents of the array items inside of message.data should be tested they are given as `any`

Reply:
Like I said, I had logged what it was sometimes it wasnt an array and it was spamming my console.. i can see if I can find the error

Reply:
`
Set `_legacyUndefinedCheck: true` on element class to enable.
[main][error] Electron event WebContents#did-fail-load observed {
  errorCode: -3,
  errorDescription: '',
  validatedURL: 'http://localhost:3000/main_window',
  isMainFrame: true,
  frameProcessId: 4,
  frameRoutingId: 1
}
Unhandled electron-log error TypeError: dataString.replaceAll is not a function
`

Reply:
Are you able to debug and see when this happens to explain what is triggering this?

Otherwise I feel like something else is at fault here? Considering the `let dataString: string = message.data[i] as string;` should require it to become a non-nullable string and therefore provide the function to be available. 

I cannot see where this would actually trigger a problem 😖
And considering I've checked both my Personal + Laptop's Logs and yet to find this error to occur on either, I suspect this might be more a fault of your Fork at the moment.